compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp

```diff
@@ -57,6 +68,55 @@ using namespace llvm;
 using namespace llvm::sys;
 using namespace llvm::object;
 
+static Error writeFile(StringRef Filename, StringRef Data) {
+  Expected<std::unique_ptr<FileOutputBuffer>> OutputOrErr =
+      FileOutputBuffer::create(Filename, Data.size());
+  if (!OutputOrErr)
+    return OutputOrErr.takeError();
+  std::unique_ptr<FileOutputBuffer> Output = std::move(*OutputOrErr);
+  llvm::copy(Data, Output->getBufferStart());
+  if (Error E = Output->commit())
+    return E;
+  return Error::success();
+}
+
+extern "C" bool rustBundleImages() {
+  SmallVector<char, 1024> BinaryData;
+  raw_svector_ostream OS(BinaryData);
+  BumpPtrAllocator Alloc;
+  StringSaver Saver(Alloc);
+
+  DenseMap<StringRef, StringRef> Args;
+  Args["triple"] = "amdgcn-amd-amdhsa"; // required
+  Args["arch"]   = "gfx90a"; // optional, but from rustc?
+
+  {
+    OffloadBinary::OffloadingImage ImageBinary{};
+
+    llvm::ErrorOr<std::unique_ptr<llvm::MemoryBuffer>> ObjectOrErr =
+        llvm::MemoryBuffer::getFileOrSTDIN("device.bc");
```

Comment:
Just looking into directly passing in the LLVMModuleRef instead of jumping through the file since we otherwise won't need it.